app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,222 @@
+  | UpdateDataDeletionTaskStatusAction
+  | SetHasMetaMetricsDataRecordedAction;
+
+// Type for the messenger of MetaMetricsDataDeletionController
+export type MetaMetricsDataDeletionControllerMessenger =
+  RestrictedControllerMessenger<
+    typeof controllerName,
+    MetaMetricsDataDeletionControllerMessengerActions,
+    never,
+    never,
+    never
+  >;
+
+/**
+ * Controller responsible for maintaining
+ * state related to Metametrics data deletion
+ */
+export class MetaMetricsDataDeletionController extends BaseController<
+  typeof controllerName,
+  MetaMetricsDataDeletionState,
+  MetaMetricsDataDeletionControllerMessenger
+> {
+  #dataDeletionService: PublicInterface<DataDeletionService>;
+
+  #getMetaMetricsId: () => string | null;
+
+  /**
+   * Creates a MetaMetricsDataDeletionController instance.
+   *
+   * @param args - The arguments to this function.
+   * @param args.dataDeletionService - The service used for deleting data.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
+   * @param args.getMetaMetricsId - A function that returns the current MetaMetrics ID.
+   */
+  constructor({
+    dataDeletionService,
+    messenger,
+    state,
+    getMetaMetricsId,
+  }: {
+    dataDeletionService: PublicInterface<DataDeletionService>;
+    messenger: MetaMetricsDataDeletionControllerMessenger;
+    state?: Partial<MetaMetricsDataDeletionState>;
+    getMetaMetricsId: () => string | null;
+  }) {
+    // Call the constructor of BaseControllerV2
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { ...defaultState, ...state },
+    });
+    this.#getMetaMetricsId = getMetaMetricsId;
+    this.#dataDeletionService = dataDeletionService;
+    this.#registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering this controller's messaging system
+   * actions.
+   */
+  #registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:createMetaMetricsDataDeletionTask`,
+      this.createMetaMetricsDataDeletionTask.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:updateDataDeletionTaskStatus`,
+      this.updateDataDeletionTaskStatus.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:setHasMetaMetricsDataRecorded`,
+      this.setHasMetaMetricsDataRecorded.bind(this),
+    );
+  }
+
+  /**
+   * Creating the delete regulation using source regulation
+   *
+   */
+  async createMetaMetricsDataDeletionTask(): Promise<void> {
+    const metaMetricsId = this.#getMetaMetricsId();
+    if (!metaMetricsId) {
+      throw new Error('MetaMetrics ID not found');
+    }
+
+    const { data } =
+      await this.#dataDeletionService.createDataDeletionRegulationTask(
+        metaMetricsId,
+      );
+    this.update((state) => {
```

Comment:
We should clear `metaMetricsDataDeletionStatus` here as well. If it's set, then it must have been pertaining to a previous deletion request, so it would be inaccurate at this point.

The same goes for `hasMetaMetricsDataRecorded` as well, though in my other comment I suggest removing that state altogether.

Reply:
@NiranjanaBinoy was this comment addressed? I see that `this.updateDataDeletionTaskStatus()` is called, but it does not clear the status, rather it updates it

Reply:
Yes, now `metaMetricsDataDeletionStatus` will get updated with the current/newest request status with this call to `this.updateDataDeletionTaskStatus()`.